python/src/scipp/utils/get.py

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Gregory Tucker
+
+
+def get(x, key, default=None):
+    """
+    Get the value associated with the provided key from a `dict`-like object
+
+    :param x: dictlike
+        Any object with `__contains__()` or `__iter__()` and `__getitem__()`
+        methods which accept `key`
+    :param key:
+        The key to extract from the provided `dict`-like object`
+    :param default:
+        A default value to return if the provided key is not present
+    :return:
+        Either the value associated with the key or the default if the key
+        is missing. If he object does not implement the required methods an
+        error will be thrown.
```

Comment:
Docs are outdated now.

Reply:
The [`in` operator](https://docs.python.org/3/reference/expressions.html#in) requires that the object define `__contains__()`, `__iter__()`, or `__getitem__()` (with the last only possible if an integer value can be used as the index).

`x[key] if key in x` uses `x.__getitem__(key)` and either `x.__contains__(key)` or `x.__iter__()` since `key` is not an integer. So this function should work for any object `x` that implements the methods as indicated in the doc string.

Reply:
👍